controller/index.js

```diff
@@ -1,1019 +1,1054 @@
+  /**
+   *
+   * @param {object} req - the incoming request object
+   * @param {object} res - the outgoing response object
+   */
+  controller.list = function (req, res) {
+    agol.find(null, function (err, data) {
       if (err) {
-        res.status(500).send( err );
-      } else {
-        res.json( data );
+        return res.status(500).send(err)
       }
-    });
-  };
+      res.json(data)
+    })
+  }
 
-  // looks up a host based on a given id 
-  controller.find = function(req, res){
-    agol.find(req.params.id, function(err, data){
+  /**
+   * looks up a host based on a given id
+   *
+   * @param {object} req - the incoming request object
+   * @param {object} res - the outgoing response object
+   */
+  controller.find = function (req, res) {
+    agol.find(req.params.id, function (err, data) {
       if (err) {
-        res.status(err.code || 404).send( err );
-      } else {
-        res.json( data );
+        return res.status(err.code || 404).send(err)
       }
-    });
-  };
+      res.json(data)
+    })
+  }
 
-  // get the item metadata from the host 
-  controller.findItem = function(req, res){
-    if (req.params.format){
-      this.findItemData(req, res);
-    } else {
-      agol.find(req.params.id, function(err, data){
-        if (err) {
-          res.status(err.code || 404).send( err );
-        } else {
-          // Get the item 
-          agol.getItem( data.host, req.params.item, req.query, function(error, itemJson){
-            if (error) {
-              res.status(404).send( error);
-            } else {
-              res.contentType('text'); 
-              res.json( itemJson );
-            }
-          });
-        }
-      });
+  /**
+   * get the item metadata from the host
+   *
+   * @param {object} req - the incoming request object
+   * @param {object} res - the outgoing response object
+   */
+  controller.findItem = function (req, res) {
+    if (req.params.format) {
+      return this.findItemData(req, res)
     }
-  };
 
-  // drops the cache for an item
-  controller.dropItem = function(req, res){
-    // if we have a layer then append it to the query params 
-    if ( req.params.layer ) {
-      req.query.layer = req.params.layer;
+    agol.find(req.params.id, function (err, data) {
+      if (err) {
+        return res.status(err.code || 404).send(err)
+      }
+
+      // Get the item
+      agol.getItem(data.host, req.params.item, req.query, function (error, itemJson) {
+        if (error) {
+          return res.status(404).send(error)
+        }
+
+        res.contentType('text')
+        res.json(itemJson)
+      })
+    })
+  }
+
+  /**
+   * Drops the cache for an item
+   * @param {object} req - the incoming request object
+   * @param {object} res - the outgoing response object
+   */
+  controller.dropItem = function (req, res) {
+    // if we have a layer then append it to the query params
+    if (req.params.layer) {
+      req.query.layer = req.params.layer
```

Comment:
this is weird. why do we need the query params for a drop, shouldnt we just need the table?